### PR TITLE
changed column selection: table `derivation_paths`

### PR DESCRIPTION
**Pull request by neurosis69**
To reduce overhead and readability, in selects use column names instead of *.

Very simple example, selecting the whole row compared with the first column ( table has ~20k rows)
```
sqlite> explain select derivation_index from derivation_paths;
addr  opcode         p1    p2    p3    p4             p5  comment      
----  -------------  ----  ----  ----  -------------  --  -------------
0     Init           0     7     0                    00  Start at 7   
1     OpenRead       1     20    0     k(2,,)         00  root=20 iDb=0; derivation_index_index
2     Rewind         1     6     1     0              00               
3       Column         1     0     1                    00  r[1]=derivation_paths.derivation_index
4       ResultRow      1     1     0                    00  output=r[1]  
5     Next           1     3     0                    01               
6     Halt           0     0     0                    00               
7     Transaction    0     0     34    0              01  usesStmtJournal=0
8     Goto           0     1     0                    00               
Run Time: real 0.000 user 0.000108 sys 0.000114
```

```
sqlite> explain select * from derivation_paths;
addr  opcode         p1    p2    p3    p4             p5  comment      
----  -------------  ----  ----  ----  -------------  --  -------------
0     Init           0     13    0                    00  Start at 13  
1     OpenRead       0     19    0     7              00  root=19 iDb=0; derivation_paths
2     Rewind         0     12    0                    00               
3       Column         0     0     1                    00  r[1]=derivation_paths.derivation_index
4       Column         0     1     2                    00  r[2]=derivation_paths.pubkey
5       Column         0     2     3                    00  r[3]=derivation_paths.puzzle_hash
6       Column         0     3     4                    00  r[4]=derivation_paths.wallet_type
7       Column         0     4     5                    00  r[5]=derivation_paths.wallet_id
8       Column         0     5     6                    00  r[6]=derivation_paths.used
9       Column         0     6     7                    00  r[7]=derivation_paths.hardened
10      ResultRow      1     7     0                    00  output=r[1..7]
11    Next           0     3     0                    01               
12    Halt           0     0     0                    00               
13    Transaction    0     0     34    0              01  usesStmtJournal=0
14    Goto           0     1     0                    00               
Run Time: real 0.001 user 0.000302 sys 0.000000
```

If at all there may be very little performance improvement changing this, but still it brings more readability and less instructions. 
[How to Interpret the Bytecode](https://www.sqlite.org/opcode.html#viewing_the_bytecode)

### CODE REVIEW

**Comment by neurosis69** [comment]
@arvidn 
what are your thoughts regarding this PR and related #10323, #10368 and #10362?

On testnet10 I'm currently testing and tracing a wallet full sync with more than 32000 transactions distributed over the heights from around 65000 to the current peak, which is around 700000 .. and I see a ton of these `select * from ...` where they aren't needed, imo. Sometimes just the `wallet_id` or `wallet_type` of the coin record is used and another time it is just checked if the coin is `existing`. 

At least, that is how I read the code and trace.

File: chia/wallet/wallet_puzzle_store.py [hunk 1]
```diff
@@ -332,13 +340,13 @@ async def get_all_puzzle_hashes(self) -> Set[bytes32]:
...
 
-        cursor = await self.db_connection.execute("SELECT * from derivation_paths")
+        cursor = await self.db_connection.execute("SELECT puzzle_hash FROM derivation_paths")
         rows = await cursor.fetchall()
         await cursor.close()
         result: Set[bytes32] = set()
 
         for row in rows:
-            result.add(bytes32(bytes.fromhex(row[2])))
+            result.add(bytes32(bytes.fromhex(row[0])))
```
Comment (arvidn) [inline chia/wallet/wallet_puzzle_store.py:349]: it's odd that this doesn't say `bytes32.fromhex()`, but that's pre-existing

**Review by arvidn — APPROVED** [review]
this all looks correct, and is definitely an improvement!

**Review by wjblanke — APPROVED** [review]
aok

**Pull request merged**